### DISCUSSION TITLE
issue with quic

### DISCUSSION THREAD

**Discussion by hellgod77** [ZoeyVid/NPMplus#795] [Q&A] [answered]
2024/04/29 16:01:27 [error] 1136#1136: *999 quic getsockopt(SO_COOKIE) failed (92: Protocol not available) while creating quic connection, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443
2024/04/29 16:01:27 [error] 1136#1136: *999 quic bpf failed to generate socket key while creating quic connection, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443
2024/04/29 16:01:27 [error] 1136#1136: *999 quic getsockopt(SO_COOKIE) failed (92: Protocol not available) while handling frames, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443

the ip listed is not the origin, but the client.. so is there a module missing from the distro, or client issue?

**Comment by Zoey2936** [comment]
which client do you use?

**Comment by hellgod77** [comment]
Edge, and its trying to proxy to npmplus so the ip listed above is connecting to origin xxx.xxx.com and its using npm to it's self 127.0.0.1:81 this is on centos 9 the rest i used are on Debian so i was just wondering if its a os issue
________________________________

**Comment by Zoey2936** [comment]
what kernel do you run?

**Comment by Zoey2936** [comment]
When I test quic/http3 I always use my curl build with http3/quic support: https://github.com/ZoeyVid/curl-quic maybe you can try it yourself with it?

**Comment by hellgod77** [comment]
i use edge browser if thats wat your asking


"speedtest.aay.limo", referrer: "https://speedtest.aay.limo/"

2024/04/29 17:01:36 [alert] 3355#3355: *3491 sendmsg() failed (90: Message too large) while sending frames, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443

2024/04/29 17:01:42 [error] 3344#3344: *3508 quic getsockopt(SO_COOKIE) failed (92: Protocol not available) while creating quic connection, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443

2024/04/29 17:01:42 [error] 3344#3344: *3508 quic bpf failed to generate socket key while creating quic connection, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443

2024/04/29 17:01:42 [error] 3344#3344: *3508 quic getsockopt(SO_COOKIE) failed (92: Protocol not available) while handling frames, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443

2024/04/29 17:01:42 [error] 3344#3344: *3508 quic bpf failed to generate socket key while handling frames, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443

2024/04/29 17:01:42 [alert] 3344#3344: *3508 sendmsg() failed (90: Message too large) while sending frames, client: 2600:6c40:1300:9a00:940b:4cad:8d3b:6223, server: [::]:443

thats what im getting regardless of what its proxying im am goin to try a differnt pc in a bit but the other ones on debian dont do that on this client

________________________________

**Comment by Zoey2936** [comment]
no, I mean if you can download the curl binary and connect to your NPMplus

**Comment by hellgod77** [comment]
im still a newb as much i know about curl is it works like wget
________________________________

**Comment by Zoey2936** [comment]
you can also give my the domain name you open if it is public, then I test myself

**Comment by hellgod77** [comment]
speedtest.aay.limo

its not much i just use it for testing so public is fine if you need access t docker or portainer that would need to be private

________________________________

**Comment by hellgod77** [comment]
On that particular system it's whatever the base install of Alma Linux 9 comes with it's on a VPS I've got on a Colo like I said that is not the normal one I use I normally use debian 12, or ubuntu 23.10. that's just what this particular host defaulted to I can change it, it ain't got anything on it yet so if I want to I can I just didn't know if you knew about that or if it was an easy fix or what cuz I've never seen it but this is also the first time I ran it on that particular operating system I have ran it on the actual centos stream, but I don't remember if it had any issues like that or not because I only used it to keep free PBX off the main internet and be able to get to the interface. Also I've been meaning to ask you I noticed in like the settings for the SSL did it has options for cpanel and GoDaddy etc can it automatically add the DNS to those providers or does it just use it to verify the domain. Because you had a way to automatically add them that would be epic it would save so much time but it is just using it as a way to verify then it's really not much different than just letting it use port 80 unless it's set up somewhere for 80 isn't available but then that would kind of defeat the purpose of using the entire programOn Apr 29, 2024 4:12 PM, Zoey ***@***.***> wrote:
what kernel do you run?

**Comment by hellgod77** [comment]
Would that be installed on the same machine that npm is? I'm pretty sure there's a dependency missing and that's why it's doing it I just don't know what oneOn Apr 29, 2024 4:14 PM, Zoey ***@***.***> wrote:
When I test quic/http3 I always use my curl build with http3/quic support: https://github.com/ZoeyVid/curl-quic maybe you can try it yourself with it?

**Comment by Zoey2936** [comment, marked as answer]
http3/quic works without problems:
```
PC-Zoey:~# ./curl-aarch64 --http3-only -I https://speedtest.aay.limo
HTTP/3 200
date: Tue, 30 Apr 2024 12:53:25 GMT
content-type: text/html
content-length: 8020
vary: accept-encoding
last-modified: Sun, 28 Jan 2024 09:04:38 GMT
vary: Accept-Encoding
etag: "65b618a6-1f54"
expires: Wed, 30 Apr 2025 12:53:25 GMT
cache-control: max-age=31536000
cache-control: public
vary: Accept-Encoding
accept-ranges: bytes
alt-svc: h3=":443"; ma=86400
```

**Reply by gomaaz** [reply]
I checked my website also...with success but usual online checker like https://http3check.net tells me its not quic...?

**Reply by Zoey2936** [reply]
New or migrated instance?

**Reply by gomaaz** [reply]
New.

I found something out. 

When the subdomain is proxied through Cloudflare
curl test - pass
http3check - fail

if it's not proxied
curl - pass
http3check - pass

http3 on cloudflare is activated.

**Reply by Zoey2936** [reply]
If you proxy npmplus through cloudflare then you need to report the error to cloudflare
